src/docs/product/issues/issue-details/performance-issues/main-thread-io.mdx

```diff
@@ -0,0 +1,72 @@
+---
+title: "IO on Main Thread for Mobile"
+sidebar_order: 20
+description: "Learn more about File IO on Main Thread and how to diagnose and fix these issues."
+---
+
+<Note>
+
+This feature is available only if you're in the Early Adopter program, and have the latest version of either the Android or iOS Mobile SDK. Features available to Early Adopters are still in-progress and may have bugs. We recognize the irony.
+
+If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily. You can find information about the previous version of reprocessing in [Legacy Reprocessing](#legacy-reprocessing).
+
+</Note>
+
+An important concept for mobile application's is the Main UI thread, the thread in charge of handling events such as button presses, or scrolling the page. This means that long running operations like File IO on the main thread will cause the User Interface to block until the operation is complete. This can be a significant problem if it causes your application to not respond
```

Comment:
```suggestion
The Main UI thread in a mobile application handles user interface events such as button presses and page scrolls. If the Main UI thread performs long running operations like File IO, it will block the whole UI until the operation is complete, preventing the user from interacting with the app. If the Main UI thread is blocked for a long period of time, it can cause App Hangs and Application Not Responding errors.
```